processing/src/main/java/org/apache/druid/segment/nested/CompressedNestedDataComplexColumn.java

```diff
@@ -534,9 +537,14 @@ public ColumnValueSelector<?> makeColumnValueSelector(List<NestedPathPart> path,
       if (arrayFieldIndex >= 0) {
         final int elementNumber = ((NestedPathArrayElement) lastPath).getIndex();
         if (elementNumber < 0) {
-          throw new IAE("Cannot make array element selector for path [%s], negative array index not supported for this selector", path);
+          throw new IAE(
+              "Cannot make array element selector for path [%s], negative array index not supported for this selector",
+              path);
```

Comment:
Bit of a nit, but I just noticed that this is still throw IAE, it would probably be good to switch these to user-facing DruidExceptions as I think that all of these usages are expecting to get back to the user.